libmachine/persist/secretstore.go

```diff
@@ -0,0 +1,210 @@
+package persist
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rancher/machine/libmachine/host"
+	"github.com/rancher/machine/libmachine/log"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1types "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type SecretStore struct {
+	Store
+	SecretName, SecretNamespace string
+	SecretClient                corev1types.SecretInterface
+	secret                      *v1.Secret
+}
+
+func NewSecretStore(store Store, secretName, secretNamespace, kubeConfigPath string) (Store, error) {
+	var config *rest.Config
+	var err error
+	if kubeConfigPath != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: Pull data from secret and create directories
+	s := &SecretStore{Store: store, SecretName: secretName, SecretNamespace: secretNamespace, SecretClient: clientset.CoreV1().Secrets(secretNamespace)}
+	err = s.extractConfig()
+	if err != nil {
```

Comment:
nit (applies to any where this is done unless this format would make the line unreasonable long):
```suggestion
	if err = s.extractConfig(); err != nil {
```

Reply:
Is this the preferred way because it is easier to read?